### ISSUE TITLE
conflict with redmine_CKEditor

### ISSUE THREAD

**Issue by juxta73** [akiko-pusu/redmine_banner#111]
hi, i have conflict with redmine_CKEditor.
if remove redmine_baner, then the error in ckeditor does not appear. 

same details (https://github.com/a-ono/redmine_ckeditor/issues/306).

Log:

Started GET "/settings/plugin/redmine_ckeditor" for 127.0.0.1 at 2019-07-29 14:49:58 +0800
Processing by SettingsController#plugin as HTML
Parameters: {"id"=>"redmine_ckeditor"}
Current user: admin (id=1)
Rendering settings/plugin.html.erb within layouts/admin
Rendered plugins/redmine_ckeditor/app/views/settings/_ckeditor.html.erb (411.5ms)
Rendered settings/plugin.html.erb within layouts/admin (425.3ms)
Completed 500 Internal Server Error in 495ms (ActiveRecord: 25.3ms)

ActionView::Template::Error (undefined local variable or method `ckeditor_javascripts' for #<#Class:0x000000000c2f81c0:0x000000000c2f1e38>):
1: <%= ckeditor_javascripts %>
2: <%= stylesheet_link_tag 'editor', :plugin => 'redmine_ckeditor'%>
3: <%= stylesheet_link_tag 'selector', :plugin => 'redmine_ckeditor'%>
4:

plugins/redmine_ckeditor/app/views/settings/_ckeditor.html.erb:1:in _plugins_redmine_ckeditor_app_views_settings__ckeditor_html_erb__87485380_102164980' app/views/settings/plugin.html.erb:6:in block in _app_views_settings_plugin_html_erb__39754121_102220920'
app/views/settings/plugin.html.erb:4:in _app_views_settings_plugin_html_erb__39754121_102220920' lib/redmine/sudo_mode.rb:63:in sudo_mode'

Redmine plugins:

redmine_banner: 0.2.0
redmine_ckeditor: 1.2.2

**Comment by akiko-pusu** [comment]
@juxta73 
Hi, thanks for using my plugin. I’ll confirm above.

**Comment by akiko-pusu** [comment]
@juxta73 
I can reproduce and confirm the problem. Thanks for your reporting!

This is because some workaround to prevent conflict with another plugin which previous version applied is missing on latest version.

Here is my workaround for this issue.
Could you please try? Of course I’m going to release bug fix version soon.

```patch
diff --git a/init.rb b/init.rb
index fbb1c22..79ec564 100644
--- a/init.rb
+++ b/init.rb
@@ -4,7 +4,7 @@ require 'banners/settings_controller_patch'
 require 'banners/projects_helper_patch'
 
 # NOTE: Keep error message for a while to support Redmine3.x users.
-def issue_template_version_message(original_message = nil)
+def banner_version_message(original_message = nil)
   <<-"USAGE"
   ==========================
   #{original_message}
@@ -44,7 +44,13 @@ Redmine::Plugin.register :redmine_banner do
     project_module :banner do
       permission :manage_banner, { banner: %I[show edit project_banner_off] }, require: :member
     end
+
+    Rails.configuration.to_prepare do
+      unless SettingsController.included_modules.include?(Banners::SettingsControllerPatch)
+        SettingsController.send(:prepend, Banners::SettingsControllerPatch)
+      end
+    end
   rescue ::Redmine::PluginRequirementError => e
-    raise ::Redmine::PluginRequirementError.new(issue_template_version_message(e.message)) # rubocop:disable Style/RaiseArgs
+    raise ::Redmine::PluginRequirementError.new(banner_version_message(e.message)) # rubocop:disable Style/RaiseArgs
   end
 end
diff --git a/lib/banners/banner_helper.rb b/lib/banners/banner_helper.rb
index e9343fb..573d4cb 100644
--- a/lib/banners/banner_helper.rb
+++ b/lib/banners/banner_helper.rb
@@ -1,7 +1,11 @@
 module Banners
   module BannerHelper
     def get_time(ymd, h, m)
-      d = Date.strptime(ymd, '%Y-%m-%d')
+      begin
+        d = Date.strptime(ymd, '%Y-%m-%d')
+      rescue
+        d = Date.current
+      end
       Time.mktime(d.year, d.month, d.day, h.to_i, m.to_i)
     end
 
diff --git a/lib/banners/settings_controller_patch.rb b/lib/banners/settings_controller_patch.rb
index 0b5aefb..9e074c4 100644
--- a/lib/banners/settings_controller_patch.rb
+++ b/lib/banners/settings_controller_patch.rb
@@ -91,4 +91,3 @@ module Banners
   end
 end
 
-SettingsController.prepend Banners::SettingsControllerPatch
```

**Comment by juxta73** [comment]
great work! it work perfectly
thanks you

**Comment by akiko-pusu** [comment]
I’ve just released v0.2.1, including above bugfix. 
Thanks for your supporting :star2:
